// custom_vue/Maple/base.js
const HtmlWebpackPlugin = require('html-webpack-plugin');
const StylelintPlugin = require('stylelint-webpack-plugin');
const {
    VueLoaderPlugin
} = require('vue-loader');
const utils = require('./utils');

module.exports = {
    entry: {
        app: utils.resolve('./src/index.js')
    },
    resolve: {
        extensions: ['.js', '.vue', '.json'],
        alias: {
            vue$: 'vue/dist/vue.esm.js',
            '@': utils.resolve('src')
        }
    },
    module: {
        rules: [{
            test: /\.(js|vue)$/,
            loader: 'eslint-loader',
            enforce: 'pre',
            options: {
                formatter: require('eslint-friendly-formatter'),
                emitWarning: true,
                failOnError: false
            }
        },
        {
            test: /\.js$/,
            include: [utils.resolve('src')],
            use: {
                loader: 'babel-loader',
                options: {
                    presets: ["@babel/preset-env"],
                    plugins: ["@babel/plugin-transform-runtime", "@babel/plugin-proposal-class-properties", "transform-vue-jsx"]
                }
            }
        },
        {
            test: /\.(png|jpe?g|gif|svg)(\?.*)?$/,
            loader: 'url-loader',
            options: {
                limit: 10000,
                name: utils.assetsPath('img/[name].[hash:7].[ext]'),
                esModule: false
            },
            include: [utils.resolve('src'), utils.resolve('static')]
        },
        {
            test: /\.(mp4|webm|ogg|mp3|wav|flac|aac)(\?.*)?$/,
            loader: 'url-loader',
            options: {
                limit: 10000,
                name: utils.assetsPath('media/[name].[hash:7].[ext]'),
                esModule: false
            },
            include: [utils.resolve('src'), utils.resolve('static')]
        },
        {
            test: /\.(woff2?|eot|ttf|otf)(\?.*)?$/,
            loader: 'url-loader',
            options: {
                limit: 10000,
                name: utils.assetsPath('fonts/[name].[hash:7].[ext]'),
                esModule: false
            },
            include: [utils.resolve('src'), utils.resolve('static')]
        }, {
            test: /\.vue$/,
            loader: 'vue-loader',
            include: [utils.resolve('src')]
        }
        ]
    },
    plugins: [
        new VueLoaderPlugin(),
        new HtmlWebpackPlugin({
            filename: 'index.html',
            template: 'index.html',
            inject: true
        }),
        new StylelintPlugin({
            files: ['**/*.{html,vue,css,sass,scss,less}'],
            fix: false,
            cache: false,
            emitWarning: true,
            failOnError: true
        })
    ]
};
