// custom_vue/Maple/build.js
const merge = require('webpack-merge');
const CopyWebpackPlugin = require('copy-webpack-plugin');
const {
    CleanWebpackPlugin
} = require('clean-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const baseWebpackConfig = require('./base');
const utils = require('./utils');

module.exports = merge(baseWebpackConfig, {
    mode: 'production',
    output: {
        filename: utils.assetsPath('js/[name].[chunkhash].js'),
        chunkFilename: utils.assetsPath('js/[id].[chunkhash].js')
    },
    module: {
        rules: [{
            test: /\.css$/,
            use: [
                MiniCssExtractPlugin.loader,
                'css-loader',
                utils.postcssLoader
            ]
        }, {
            test: /\.less$/,
            use: [
                MiniCssExtractPlugin.loader,
                'css-loader',
                'less-loader',
                utils.postcssLoader
            ]
        }, {
            test: /\.scss$/,
            use: [
                MiniCssExtractPlugin.loader,
                'css-loader',
                'sass-loader',
                utils.postcssLoader
            ]
        }]
    },
    plugins: [
        new CleanWebpackPlugin(),
        new CopyWebpackPlugin([{
            from: utils.resolve('./static'),
            to: 'static',
            ignore: ['.*']
        }]),
        new MiniCssExtractPlugin({
            filename: utils.assetsPath('css/[name].[chunkhash].css')
        })
    ]
});
