// custom_vue/Maple/dev.js
const merge = require('webpack-merge');
const webpack = require('webpack');
const baseWebpackConfig = require('./base');
const utils = require('./utils');
module.exports = merge(baseWebpackConfig, {
    mode: 'development',
    devtool: 'inline-source-map',
    devServer: {
        port: 1208,
        host: '127.0.0.1',
        open: true,
        hot: true,
        overlay: {
            warnings: false,
            errors: true
        }
    },
    module: {
        rules: [{
            test: /\.css$/,
            use: ['style-loader',
                {
                    loader: 'css-loader',
                    options: {
                        sourceMap: false
                    }
                },
                utils.postcssLoader
            ]
        },
        {
            test: /\.less$/,
            use: ['style-loader',
                {
                    loader: 'css-loader',
                    options: {
                        sourceMap: false
                    }
                },
                {
                    loader: 'less-loader',
                    options: {
                        sourceMap: false
                    }
                },
                utils.postcssLoader
            ]
        },
        {
            test: /\.scss$/,
            use: ['style-loader',
                {
                    loader: 'css-loader',
                    options: {
                        sourceMap: false
                    }
                },
                {
                    loader: 'sass-loader',
                    options: {
                        sourceMap: false
                    }
                },
                utils.postcssLoader
            ]
        }
        ]
    },
    plugins: [
        new webpack.HotModuleReplacementPlugin()
    ]
});
